_posts/2023-07-10-custom-http-signaling-notification.adoc

```diff
@@ -0,0 +1,252 @@
+
+The complete source code for this example in the blog post is provided in the Debezium examples repository under https://github.com/debezium/debezium-examples/tree/main/http-signaling-notification[http-signal-notification] directory.
+
+Create a java project to build the `HTTP` signal and notification channels. Run the following command to create a new java project using Maven:
+
+[source,shell]
+----
+mvn archetype:generate
+    -DgroupId=io.debezium.examples
+    -DartifactId=http-signaling-notification
+----
+
+Add the following dependency to the `pom.xml` file with the Debezium version (2.3 and later versions):
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.debezium</groupId>
+    <artifactId>debezium-core</artifactId>
+    <version>${debezium.version}</version>
+</dependency>
+----
+
+To receive signals using a Mock Server, create a Docker Compose file that defines the Mock Server service. The configuration for the Mock Server service will be as follows:
+
+[source,yaml]
+----
+services:
+  mockServer:
+    image: mockserver/mockserver:latest
+    ports:
+      - 1080:1080
+    environment:
+      - MOCKSERVER_WATCH_INITIALIZATION_JSON=true
+      - MOCKSERVER_INITIALIZATION_JSON_PATH=/config/initializerJson.json
+    volumes:
+        - ./initializerJson.json:/config/initializerJson.json
+----
+
+The environment variables `MOCKSERVER_WATCH_INITIALIZATION_JSON` and `MOCKSERVER_INITIALIZATION_JSON_PATH` are set to enable the Mock Server to watch for changes in the initialization JSON file and specify its path. The `initializerJson.json` file, which contains the http request and response information for the signals, is mounted into the Mock Server container. The definition of the `initializerJson.json` file is as follows:
+
+[source,json]
+----
+[
+  {
+    "httpRequest" : {
+      "method" : "GET",
+      "path" : "/api/signal",
+      "queryStringParameters" : {
+        "code" : ["10969"]
+      }
+    },
+    "httpResponse" : {
+      "body": "{\"id\":\"924e3ff8-2245-43ca-ba77-2af9af02fa07\",\"type\":\"log\",\"data\":{\"message\": \"Signal message received from http endpoint.\"}}",
+      "statusCode": 200
+    }
+  }
+]
+----
+
+The `initializerJson.json` file defines a mock http request to the path `/api/signal` with the query string parameter `code=10969`. When this request is received by the Mock Server, it will respond with a JSON body containing an `id`, `type`, and `data`. The response has a status code of 200, indicating a successful response.
+
+The `id` is an arbitrary unique string that identifies a signal instance.
+The `type` specifies the type of signal to send. In this example, the type is `log`. It requests the connector to add an entry to the log. After processing the signal, the connector prints the specified message to the log.
+The `data` specifies JSON-formatted parameters to pass to a signal event. In this example, the `message` parameter is passed to the signal event.
```

Comment:
```suggestion
`id`:: an arbitrary unique string that identifies a signal instance.
`type`:: the type of signal to send. In this example, the type is `log`, and it requests the connector to add an entry to the connector's log file. After the signal is processed, the connector prints the specified message in the log.
`data`:: the JSON-formatted parameters to pass to a signal event. In this example, the `message` parameter is passed to the signal event.
```
We'll use description lists as they tend to be easier to read after a code segment.  If you'd prefer to use the style in the two code blocks below, that's also acceptable.  I'd suggest whichever style you pick, just be consistent in all areas.